### PR TITLE
Update COM type descriptor extended browsing handlers

### PR DESCRIPTION
**Pull request by JeremyKuhne**
This cleans up the COM type descriptor extended property handlers for supported property interfaces.

- Uses interfaces directly, enabling ComWrapper scenarios
- Simplifies the handler instance handling - no longer uses Activator.CreateInstance
- Bring variable names up to standard (notably for event args)
- Add simple supports interface helper in ComHelpers
- Add interface for COM2ExtendedBrowsingHandler and make base class generic of interface type

Additional PRs will follow that change away from casting to managed COM interfaces to suppport AOT / ComWrappers.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/9244)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IProvidePropertyBuilderHandler.cs [hunk 1]
```diff
@@ -5,83 +5,83 @@
...
-        string? guidString = null;
-        CTLBLDTYPE bldrType = 0;
-        bool builderValid = GetBuilderGuidString(target, sender.DISPID, ref guidString, &bldrType);
+        string? builderGuid = null;
+        CTLBLDTYPE builderType = 0;
+        bool builderValid = GetBuilderGuidString(propertyBuilder, sender.DISPID, ref builderGuid, &builderType);
 
-        // We hide IDispatch props by default, we we need to force showing them here
-        if (sender.CanShow && builderValid && typeof(Oleaut32.IDispatch).IsAssignableFrom(sender.PropertyType))
+        // We hide IDispatch props by default, we we need to force showing them here.
```
Comment (Shyam-Gupta) [inline src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IProvidePropertyBuilderHandler.cs:65]: nit: one 'we' can be removed.

File: src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IProvidePropertyBuilderHandler.cs [hunk 2]
```diff
@@ -5,83 +5,83 @@
...
+        if (sender.CanShow && builderValid)
```
Comment (Shyam-Gupta) [inline src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IProvidePropertyBuilderHandler.cs:66]: `typeof(Oleaut32.IDispatch).IsAssignableFrom(sender.PropertyType)` check is not required anymore?

File: src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IProvidePropertyBuilderHandler.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IProvidePropertyBuilderHandler.cs:65]: Ah, missed this one.

File: src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2TypeInfoProcessor.cs [hunk 3]
```diff
@@ -90,40 +93,47 @@ private static ModuleBuilder ModuleBuilder
...
-                    if (classInfo.GetInfoOfIndex(
+                    if (classInfo.Value->GetInfoOfIndex(
                         i,
                         MULTICLASSINFO_FLAGS.MULTICLASSINFO_GETTYPEINFO,
                         &typeInfo,
                         pdwTIFlags: null,
                         pcdispidReserved: null,
                         piidPrimary: null,
-                        piidSource: null).Failed)
+                        piidSource: null).Succeeded
```
[the remark this answers is not in the document]
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2TypeInfoProcessor.cs:118]: Good catch- I must have undone here. I'm going to write some tests but I can't until I finish this series of changes. :/

File: src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IProvidePropertyBuilderHandler.cs (hunk 2 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IProvidePropertyBuilderHandler.cs:66]: It is always IDispatch :)

**Review by Tanya-Solyanik — APPROVED** [review]

**Pull request merged**